### PR TITLE
Add test cases to Assign Pods to Node using nodeName field

### PR DESCRIPTION
**Pull request by sidhant-agrawal**
Added tests/manage/pv_services/test_pvc_assign_node_pod.py
 - Test to assign pods to a node using nodeName field of PodSpec
 - Automates OCS-717, 744, 1258 

Signed-off-by: Sidhant Agrawal <sagrawal@redhat.com>

### CODE REVIEW

**Comment by sidhant-agrawal** [comment]
```
======================================================== short test summary info ========================================================
PASSED tests/manage/pv_services/test_pvc_assign_node_pod.py::TestNodeAssign::test_pvc_assign_node_pod[CephBlockPool]
PASSED tests/manage/pv_services/test_pvc_assign_node_pod.py::TestNodeAssign::test_pvc_assign_node_pod[CephFileSystem]
====================================================== 2 passed in 175.10 seconds =======================================================
```
[test_pvc_assign_node_pod.log](https://github.com/red-hat-storage/ocs-ci/files/3494239/test_pvc_assign_node_pod.log)

File: tests/manage/pv_services/test_pvc_assign_node_pod.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            ),
+            pytest.param(
+                *[constants.CEPHFILESYSTEM],
+                marks=pytest.mark.polarion_id("OCS-744")
+            )
+        ]
+    )
+    def test_pvc_assign_node_pod(self, interface, pvc_factory, resources):
+        """
+        Test assign nodeName to a pod using RWO pvc
```
Comment (hnallurv) [inline tests/manage/pv_services/test_pvc_assign_node_pod.py]: Do you have test cases for RWX? if not, please consider

File: tests/manage/pv_services/test_pvc_assign_node_pod.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            namespace=pvc_obj.namespace, node_name=worker_nodes_list[0],
+            pod_dict_path=constants.NGINX_POD_YAML
+        )
+        pods.append(pod_obj)
+        assert helpers.wait_for_resource_state(
+            resource=pod_obj, state=constants.STATUS_RUNNING, timeout=120
+        )
+
+        # Reload the new information and verify the nodeName
+        pod_obj.reload()
```
Comment (hnallurv) [inline tests/manage/pv_services/test_pvc_assign_node_pod.py]: you can run a quick IO.

File: tests/manage/pv_services/test_pvc_assign_node_pod.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        assert worker_nodes_list[0] == pod_obj.pod_data.get('spec').get('nodeName')
```
Comment (ebenahar) [inline tests/manage/pv_services/test_pvc_assign_node_pod.py]: Please add an assertion message here

Comment (ebenahar) [inline tests/manage/pv_services/test_pvc_assign_node_pod.py]: Please add a log describing we are checking the pod should start on the specific node

File: tests/manage/pv_services/test_pvc_assign_node_pod.py (hunk 2 above)
Comment (sidhant-agrawal) [reply tests/manage/pv_services/test_pvc_assign_node_pod.py]: ack, will add fio for 30 sec.

File: tests/manage/pv_services/test_pvc_assign_node_pod.py (hunk 3 above)
Comment (sidhant-agrawal) [reply tests/manage/pv_services/test_pvc_assign_node_pod.py]: ack

Comment (sidhant-agrawal) [reply tests/manage/pv_services/test_pvc_assign_node_pod.py]: ack

File: tests/manage/pv_services/test_pvc_assign_node_pod.py (hunk 1 above)
Comment (sidhant-agrawal) [reply tests/manage/pv_services/test_pvc_assign_node_pod.py]: Added CephFS RWX test.

**Comment by sidhant-agrawal** [comment]
```
======================================================== short test summary info ========================================================
PASSED tests/manage/pv_services/test_pvc_assign_pod_node.py::TestPvcAssignPodNode::test_rwo_pvc_assign_pod_node[CephBlockPool]
PASSED tests/manage/pv_services/test_pvc_assign_pod_node.py::TestPvcAssignPodNode::test_rwo_pvc_assign_pod_node[CephFileSystem]
PASSED tests/manage/pv_services/test_pvc_assign_pod_node.py::TestPvcAssignPodNode::test_rwx_pvc_assign_pod_node[CephFileSystem]
```
[test_pvc_assign_pod_node.log](https://github.com/red-hat-storage/ocs-ci/files/3519724/test_pvc_assign_pod_node.log)

**Pull request merged**